bsdlib/doc/security_tags.rst

```diff
@@ -0,0 +1,51 @@
+.. _security_tags:
+
+Security tags
+#############
+
+The modem on an nRF9160 device is equipped with a full IPv4/IPv6 stack and a DTLS/TLS engine.
+To use the cryptographic functions in the modem, the application must provision the security credentials to the modem.
+
+To be able to provision credentials, the modem must be offline or in offline mode.
+The credentials are provisioned through AT commands (see `Credential storage management %CMNG <https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/mob_termination_ctrl_status/cmng.html>`_).
+If your device is running an nRF Connect SDK application, you can also use LTE Link Monitor to `manage the credentials <https://infocenter.nordicsemi.com/topic/ug_link_monitor/UG/link_monitor/lm_certificate_manager.html>`_.
+
+Each set of credentials is identified by a security tag (``sec_tag``), which is referenced when a DTLS/TLS socket is created.
+The security tag is attached to a socket using the ``setsockopt`` function before connecting (through TCP) or transferring data (through UDP).
+
+The following code snippet and figure show how to use a security tag in a TLS application::
+
+   s = socket(NRF_AF_INET, NRF_SOCK_STREAM, NRF_SPROTO_TLS1v2)
+   s.setsockopt(NRF_SOL_SECURE, NRF_SO_SEC_TAG_LIST, [1], 4)
+   s.setsockopt(NRF_SOL_SECURE, NRF_SO_PEER_VERIFY, 2, 4) # 2 = strict
+   ...
+   s.connect() # TLS handshake using sec_tag 1
+   s.send() # send encrypted data using sec_tag 1
+   ...
```

Comment:
Let's have a snippet in C code:
```
/* Let `fd` be a valid UDP or TCP socket descriptor */

int err;
int verify;
sec_tag_t sec_tag_list[] = { sec_tag };

enum {
	NONE = 0,
	OPTIONAL = 1,
	REQUIRED = 2,
};

verify = REQUIRED;

err = nrf_setsockopt(fd, NRF_SOL_TLS, NRF_TLS_PEER_VERIFY, &verify, sizeof(verify));
if (err) {
	/* Failed to setup peer verification */
	return -1;
}

err = nrf_setsockopt(fd, NRF_SOL_TLS, NRF_TLS_SEC_TAG_LIST, sec_tag_list, sizeof(sec_tag_t) * ARRAY_SIZE(sec_tag_list));
if (err) {
	/* Failed to setup socket security tag */
	return -1;
}
```

Reply:
How does that fit together with the figures and what we are explaining? I understand what we're trying to show in the current version, but I can't spot any of those concepts in your code snippet. ;-)

Maybe we should keep as is (but change "code snippet" to "pseudo code") and add this as "real code sample" after the second figure?

Reply:
If we want to explain a concept, we are better off using English, rather than pseudo-code.
If we want to show a code snippet, it better be one that can be copy-pasted.
Perhaps I could add a couple of comments to that snippet to make it clearer?

Reply:
Added a few comments to the snippet, is it clearer @ru-fu ?